src/main/haskell/kore/src/Kore/Step/Simplification/PredicateSubstitution.hs

```diff
@@ -110,7 +110,7 @@ simplify
         traverse
             (`substitute` unifiedSubstitution)
             predicate
-    if substitutedPredicate == predicate && not first
+    if substitutedPredicate == predicate && times > 1
```

Comment:
I don't think this is the right fix. If a bool is not enough here, i.e. if the formula is not simplified in the first iteration, then there is no guarantee that one more iteration would be enough.

IIRC the main issue was caused by simplification not descending into builtins (e.g. maps), which is needed after substitutions.

If you insist on keeping this, please change the RP description with
`s/fixed a simplification bug/made an ugly hack to work around builtin simplification/`
and please provide an argument for why you prefer this hack instead of the proper fix.